### PR TITLE
Improve subtitle filename parsing

### LINKED ISSUE #7057: [Issue]: external subtitle file naming
### Please describe your bug

Not sure if this is a bug, or a documentation issue. I followed the documentation here https://jellyfin.org/docs/general/server/media/subtitles.html to name some external SRT files. All files are picked up but there are some issues around DisplayTitle.

**Forced Subtitles**
I named one SRT to be forced as per docs
`Movie.Name.2021.1080p.BluRay.H264.AAC-RARBG.forced.srt`

This is picked up by JF, but I think JF is assuming language will always be present if the forced flag is.

I changed the name to `Movie.Name.2021.1080p.BluRay.H264.AAC-RARBG.en.forced.srt` and it works as expected.
```
{
Codec: "srt",
Language: "AAC-RARBG",
localizedUndefined: "Undefined",
localizedDefault: "Default",
localizedForced: "Forced",
DisplayTitle: "AAC-RARBG - Forced",
IsInterlaced: false,
IsDefault: false,
IsForced: true,
Type: "Subtitle",
Index: 5,
Score: 121222,
IsExternal: true,
IsTextSubtitleStream: true,
SupportsExternalStream: true,
Path: "/storage/media/Videos/Movies/Movie Name (2021)/Movie.Name.2021.1080p.BluRay.H264.AAC-RARBG.forced.srt"
}
```

**Non-standard names (not languages)**
I have an SRT that is English SDH. I used this filename `Movie.Name.2021.1080p.BluRay.H264.AAC-RARBG.English - SDH.srt` File is picked up fine. For some reason the spaces are stripped from the Language/DisplayTitle attributes.

```
{
Codec: "srt",
Language: "English-SDH",
localizedUndefined: "Undefined",
localizedDefault: "Default",
localizedForced: "Forced",
DisplayTitle: "English-SDH",
IsInterlaced: false,
IsDefault: false,
IsForced: false,
Type: "Subtitle",
Index: 4,
IsExternal: true,
IsTextSubtitleStream: true,
SupportsExternalStream: true,
Path: "/storage/media/Videos/Movies/Movie Name (2021)/Movie.Name.2021.1080p.BluRay.H264.AAC-RARBG.English - SDH.srt"
}
```

Commentary subtitles. Same space stripping from a file named 
`Movie.Name.2021.1080p.BluRay.H264.AAC-RARBG.Director Commentary.srt`

```
{
Codec: "srt",
Language: "DirectorCommentary",
localizedUndefined: "Undefined",
localizedDefault: "Default",
localizedForced: "Forced",
DisplayTitle: "DirectorCommentary",
IsInterlaced: false,
IsDefault: false,
IsForced: false,
Type: "Subtitle",
Index: 2,
IsExternal: true,
IsTextSubtitleStream: true,
SupportsExternalStream: true,
Path: "/storage/media/Videos/Movies/Movie Name (2021)/Movie.Name.2021.1080p.BluRay.H264.AAC-RARBG.Director Commentary.srt"
}
```

### Jellyfin Version

10.7.7

### if other:

_No response_

### Environment

```markdown
- OS: Armbian
- Virtualization: Docker
- Clients: Browser
- Browser: Chrome 96
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

_No response_

### FFmpeg logs

_No response_

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by 1337joe**
**Changes**
- Improve fuzzy name matching to ignore: case, accents, symbols/whitespace
- Add tests for name fuzzy matching
- Put text that doesn't parse as a language in the Title field instead of using language as a label

**Issues**
Fixes #7057

### CODE REVIEW

File: tests/Jellyfin.Providers.Tests/MediaInfo/SubtitleResolverTests.cs [hunk 1]
```diff
@@ -11,6 +12,24 @@ namespace Jellyfin.Providers.Tests.MediaInfo
 {
     public class SubtitleResolverTests
     {
+        private readonly ILocalizationManager _localizationManager;
+
+        public SubtitleResolverTests()
+        {
+            var frenchCultureDto = new CultureDto()
```
Comment (crobibero) [inline tests/Jellyfin.Providers.Tests/MediaInfo/SubtitleResolverTests.cs:19]: This doesn't look like French to me :)

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.Providers/MediaInfo/SubtitleResolver.cs [hunk 2]
```diff
@@ -131,51 +135,55 @@ public SubtitleResolver(ILocalizationManager localization)
...
                         if (currentSlice.Equals(".default", StringComparison.OrdinalIgnoreCase)
                             || currentSlice.Equals(".forced", StringComparison.OrdinalIgnoreCase)
                             || currentSlice.Equals(".foreign", StringComparison.OrdinalIgnoreCase))
                         {
-                            languageSpan = languageSpan[..lastDot];
                             continue;
                         }
 
-                        languageSpan = languageSpan[(lastDot + 1)..];
-                        break;
```
Comment (cvium) [inline MediaBrowser.Providers/MediaInfo/SubtitleResolver.cs:164]: You no longer break?

Comment (1337joe) [reply MediaBrowser.Providers/MediaInfo/SubtitleResolver.cs:164]: Intentional - the old way was pulling the last non-default/forced/foreign segment between "." delimiters as the language and ignoring anything else.

Switching to using the actual title field when it's not a language means collecting all the segments that don't parse as a language or special text, not just the last one. The `lastDot < matchLength` check will keep it from pulling the end of the movie name if it's "."-separated.

**Comment by 1337joe** [comment]
Closing in favor of #7255

**Pull request closed without merging**